discovery/linode/linode.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linode
+
+import (
+	"context"
+	"bytes"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/linode/linodego"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/version"
+
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/refresh"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+)
+
+const (
+	linodeLabel              = model.MetaLabelPrefix + "linode_"
+	linodeLabelID            = linodeLabel + "instance_id"
+	linodeLabelName          = linodeLabel + "instance_label"
+	linodeLabelImage         = linodeLabel + "image"
+	linodeLabelPrivateIPv4   = linodeLabel + "private_ipv4"
+	linodeLabelPublicIPv4    = linodeLabel + "public_ipv4"
+	linodeLabelPublicIPv6    = linodeLabel + "public_ipv6"
+	linodeLabelRegion        = linodeLabel + "region"
+	linodeLabelType          = linodeLabel + "type"
+	linodeLabelStatus        = linodeLabel + "status"
+	linodeLabelTags          = linodeLabel + "tags"
+	linodeLabelGroup         = linodeLabel + "group"
+	linodeLabelHypervisor    = linodeLabel + "hypervisor"
+	linodeLabelBackups       = linodeLabel + "backups"
+	linodeLabelSpecsDisk     = linodeLabel + "specs_disk"
```

Comment:
Since they are all plain numbers, how about we append the unit to the meta label, and use standard units (bytes for all).